<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Getting Started with KDPS • KDPS: Kinship Decouple and Phenotype Selection</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="96x96" href="../favicon-96x96.png">
<link rel="icon" type="”image/svg+xml”" href="../favicon.svg">
<link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png">
<link rel="icon" sizes="any" href="../favicon.ico">
<link rel="manifest" href="../site.webmanifest">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/Fira_Sans-0.4.10/font.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Getting Started with KDPS">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-primary" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">KDPS: Kinship Decouple and Phenotype Selection</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.0.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/kdps-intro.html">Getting Started with KDPS</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav"></ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Getting Started with KDPS</h1>
            
      

      <div class="d-none name"><code>kdps-intro.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p><code>KDPS</code> (Kinship Decouple and Phenotype Selection) is an R
package that resolves cryptic relatedness in genetic studies using a
phenotype-aware approach. It removes related individuals based on
kinship or IBD scores while prioritizing the retention of subjects with
phenotypes of interest.</p>
<p>This tool is useful in GWAS and epidemiological studies where
maximizing the number of unrelated individuals with relevant traits is
essential for statistical power, especially in rare or stratified
phenotypes.</p>
<hr>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>To install the latest version from GitHub:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st"><a href="https://devtools.r-lib.org/" class="external-link">"devtools"</a></span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"devtools"</span><span class="op">)</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://devtools.r-lib.org/" class="external-link">"devtools"</a></span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st">"kdps"</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"UCSD-Salem-Lab/kdps"</span><span class="op">)</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st">"kdps"</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
<hr>
</div>
<div class="section level2">
<h2 id="example-data">Example Data<a class="anchor" aria-label="anchor" href="#example-data"></a>
</h2>
<p>This package includes two example files in <code>extdata/</code>:</p>
<div class="section level4">
<h4 id="simple_pheno-txt">
<code>simple_pheno.txt</code><a class="anchor" aria-label="anchor" href="#simple_pheno-txt"></a>
</h4>
<p>This file contains phenotypic data for individuals in the cohort.
Each row represents one individual.</p>
<table class="table">
<thead><tr class="header">
<th>Column</th>
<th>Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td><code>FID</code></td>
<td>Family ID (used for linking with kinship data)</td>
</tr>
<tr class="even">
<td><code>IID</code></td>
<td>Individual ID</td>
</tr>
<tr class="odd">
<td><code>pheno1</code></td>
<td>A binary phenotype (e.g., disease status)</td>
</tr>
<tr class="even">
<td><code>pheno2</code></td>
<td>A categorical phenotype used in prioritization</td>
</tr>
<tr class="odd">
<td><code>pheno3</code></td>
<td>A continuous trait (e.g., height or biomarker)</td>
</tr>
</tbody>
</table>
<p>Example:</p>
<table class="table">
<thead><tr class="header">
<th>FID</th>
<th>IID</th>
<th>pheno1</th>
<th>pheno2</th>
<th>pheno3</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>0</td>
<td>1001</td>
<td>DISEASED</td>
<td>DISEASED2</td>
<td>109.5</td>
</tr>
<tr class="even">
<td>0</td>
<td>1002</td>
<td>HEALTHY</td>
<td>HEALTHY</td>
<td>117.18</td>
</tr>
<tr class="odd">
<td>0</td>
<td>1003</td>
<td>HEALTHY</td>
<td>HEALTHY</td>
<td>90.41</td>
</tr>
<tr class="even">
<td>0</td>
<td>1004</td>
<td>HEALTHY</td>
<td>HEALTHY</td>
<td>95</td>
</tr>
</tbody>
</table>
<hr>
</div>
<div class="section level4">
<h4 id="simple_kinship-txt">
<code>simple_kinship.txt</code><a class="anchor" aria-label="anchor" href="#simple_kinship-txt"></a>
</h4>
<p>This file encodes pairwise relatedness between individuals based on
genome-wide genotype data.</p>
<table class="table">
<colgroup>
<col width="45%">
<col width="54%">
</colgroup>
<thead><tr class="header">
<th>Column</th>
<th>Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td><code>FID1</code></td>
<td>Family ID of individual 1</td>
</tr>
<tr class="even">
<td><code>IID1</code></td>
<td>Individual ID of individual 1</td>
</tr>
<tr class="odd">
<td><code>FID2</code></td>
<td>Family ID of individual 2</td>
</tr>
<tr class="even">
<td><code>IID2</code></td>
<td>Individual ID of individual 2</td>
</tr>
<tr class="odd">
<td><code>HetHet</code></td>
<td>Proportion of sites where both individuals are heterozygous</td>
</tr>
<tr class="even">
<td><code>IBS0</code></td>
<td>Proportion of sites with no alleles in common</td>
</tr>
<tr class="odd">
<td><code>KINSHIP</code></td>
<td>Estimated kinship coefficient (values &gt; 0.0442 typically indicate
2nd-degree or closer relationships)</td>
</tr>
</tbody>
</table>
<p>Example:</p>
<table class="table">
<thead><tr class="header">
<th>FID1</th>
<th>IID1</th>
<th>FID2</th>
<th>IID2</th>
<th>HetHet</th>
<th>IBS0</th>
<th>KINSHIP</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>0</td>
<td>1001</td>
<td>0</td>
<td>1002</td>
<td>0.037</td>
<td>0.0083</td>
<td>1</td>
</tr>
<tr class="even">
<td>0</td>
<td>1003</td>
<td>0</td>
<td>1004</td>
<td>0.046</td>
<td>0.0148</td>
<td>1</td>
</tr>
</tbody>
</table>
<hr>
</div>
</div>
<div class="section level2">
<h2 id="simple-example-resolving-relatedness-in-a-small-cohort">Simple Example: Resolving Relatedness in a Small Cohort<a class="anchor" aria-label="anchor" href="#simple-example-resolving-relatedness-in-a-small-cohort"></a>
</h2>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">kdps</span><span class="op">)</span></span>
<span></span>
<span><span class="va">phenotype_file</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"simple_pheno.txt"</span>, package <span class="op">=</span> <span class="st">"kdps"</span><span class="op">)</span></span>
<span><span class="va">kinship_file</span>   <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"simple_kinship.txt"</span>, package <span class="op">=</span> <span class="st">"kdps"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">kdps_results</span> <span class="op">=</span> <span class="fu"><a href="../reference/kdps.html">kdps</a></span><span class="op">(</span></span>
<span>  phenotype_file <span class="op">=</span> <span class="va">phenotype_file</span>,</span>
<span>  kinship_file <span class="op">=</span> <span class="va">kinship_file</span>,</span>
<span>  fuzziness <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  phenotype_name <span class="op">=</span> <span class="st">"pheno2"</span>,</span>
<span>  prioritize_high <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  prioritize_low <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  phenotype_rank <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"DISEASED1"</span>, <span class="st">"DISEASED2"</span>, <span class="st">"HEALTHY"</span><span class="op">)</span>,</span>
<span>  fid_name <span class="op">=</span> <span class="st">"FID"</span>,</span>
<span>  iid_name <span class="op">=</span> <span class="st">"IID"</span>,</span>
<span>  fid1_name <span class="op">=</span> <span class="st">"FID1"</span>,</span>
<span>  iid1_name <span class="op">=</span> <span class="st">"IID1"</span>,</span>
<span>  fid2_name <span class="op">=</span> <span class="st">"FID2"</span>,</span>
<span>  iid2_name <span class="op">=</span> <span class="st">"IID2"</span>,</span>
<span>  kinship_name <span class="op">=</span> <span class="st">"KINSHIP"</span>,</span>
<span>  kinship_threshold <span class="op">=</span> <span class="fl">0.0442</span>,</span>
<span>  phenotypic_naive <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">kdps_results</span></span></code></pre></div>
<hr>
</div>
<div class="section level2">
<h2 id="function-arguments">Function Arguments<a class="anchor" aria-label="anchor" href="#function-arguments"></a>
</h2>
<p>Key arguments for <code><a href="../reference/kdps.html">kdps()</a></code> include:</p>
<ul>
<li>
<code>phenotype_file</code>, <code>kinship_file</code>: File paths
to phenotype and kinship matrices.</li>
<li>
<code>phenotype_name</code>: The column name of the phenotype to
prioritize.</li>
<li>
<code>phenotype_rank</code>: Ordered levels from most to least
important.</li>
<li>
<code>kinship_threshold</code>: Kinship score above which subjects
are considered related.</li>
<li>
<code>fuzziness</code>: Controls tolerance when resolving complex
networks (default = 0).</li>
<li>
<code>prioritize_high</code>, <code>prioritize_low</code>: If
<code>TRUE</code>, prioritizes subjects with extreme phenotype values
(numeric).</li>
<li>
<code>phenotypic_naive</code>: If <code>TRUE</code>, phenotype info
is ignored and ties are broken randomly.</li>
</ul>
<hr>
</div>
<div class="section level2">
<h2 id="output">Output<a class="anchor" aria-label="anchor" href="#output"></a>
</h2>
<p>The output is a <code>data.frame</code> with columns:</p>
<ul>
<li>
<code>FID</code>: Family ID of the subject to remove.</li>
<li>
<code>IID</code>: Individual ID of the subject to remove.</li>
</ul>
<p>You can save this output to a text file to filter out individuals in
your downstream analysis.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/write.table.html" class="external-link">write.table</a></span><span class="op">(</span><span class="va">kdps_results</span>, file <span class="op">=</span> <span class="st">"subjects_to_remove.txt"</span>, quote <span class="op">=</span> <span class="cn">FALSE</span>, row.names <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<hr>
</div>
<div class="section level2">
<h2 id="final-notes">Final Notes<a class="anchor" aria-label="anchor" href="#final-notes"></a>
</h2>
<ul>
<li>KDPS is designed for large-scale studies like UK Biobank, with
efficient performance even on complex networks.</li>
<li>Users are encouraged to interpret results in the context of
potential <strong>collider bias</strong> introduced by phenotype-aware
filtering.</li>
<li>For large studies, consider pre-filtering unrelated individuals
using tools like <code>PLINK</code> and using KDPS for final
refinement.</li>
</ul>
<hr>
<p>For updates and source code, visit: <a href="https://github.com/UCSD-Salem-Lab/kdps%22" class="external-link">https://github.com/UCSD-Salem-Lab/kdps</a></p>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Wanjun Gu.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer>
</div>





  </body>
</html>
